pages/[network]/[chain]/pull-request.tsx

```diff
@@ -15,65 +15,138 @@ import NothingFound from "components/nothing-found";
+import { useNetwork } from "x-hooks/use-network";
 
-export default function PullRequestPage() {
-  const router = useRouter();
+interface PagePullRequestProps {
+  bounty: IssueData;
+  pullRequest: pullRequest;
+  _nextI18Next?: SSRConfig;
+}
 
+export default function PullRequestPage({ pullRequest, bounty }: PagePullRequestProps) {
   const { t } = useTranslation(["common", "pull-request"]);
 
   const [showModal, setShowModal] = useState(false);
   const [isCancelling, setIsCancelling] = useState(false);
   const [isMakingReady, setIsMakingReady] = useState(false);
-  const [pullRequest, setPullRequest] = useState<pullRequest>();
+  const [currentBounty, setCurrentBounty] = useState<IssueBigNumberData>(issueParser(bounty));
+  const [currentPullRequest, setCurrentPullRequest] = useState<pullRequest>({
+    ...pullRequest,
+    createdAt: new Date(pullRequest.createdAt),
+  });
   const [isCreatingReview, setIsCreatingReview] = useState(false);
 
   const { state, dispatch } = useAppState();
-
-  const { getDatabaseBounty } = useBounty();
+  const router = useRouter();
   const { getURLWithNetwork } = useNetwork();
   const { createReviewForPR, processEvent } = useApi();
-  const { getExtendedPullRequestsForCurrentBounty } = useBounty();
   const { handleMakePullRequestReady, handleCancelPullRequest } = useBepro();
 
   const { prId, review } = router.query;
 
   const isWalletConnected = !!state.currentUser?.walletAddress;
   const isGithubConnected = !!state.currentUser?.login;
-  const isPullRequestOpen = pullRequest?.state?.toLowerCase() === "open";
-  const isPullRequestReady = !!pullRequest?.isReady;
-  const isPullRequestCanceled = !!pullRequest?.isCanceled;
-  const isPullRequestCancelable = !!pullRequest?.isCancelable;
-  const isPullRequestCreator = pullRequest?.userAddress === state.currentUser?.walletAddress;
+  const isPullRequestOpen = currentPullRequest?.state?.toLowerCase() === "open";
+  const isPullRequestReady = !!currentPullRequest?.isReady;
+  const isPullRequestCanceled = !!currentPullRequest?.isCanceled;
+  const isPullRequestCancelable = !!currentPullRequest?.isCancelable;
+  const isPullRequestCreator = currentPullRequest?.userAddress === state.currentUser?.walletAddress;
   const branchProtectionRules = state.Service?.network?.repos?.active?.branchProtectionRules;
   const approvalsRequired =
     branchProtectionRules ?
-      branchProtectionRules[state.currentBounty?.data?.branch]?.requiredApprovingReviewCount || 0 : 0;
+      branchProtectionRules[currentBounty?.branch]?.requiredApprovingReviewCount || 0 : 0;
   const canUserApprove = state.Service?.network?.repos?.active?.viewerPermission !== "READ";
-  const approvalsCurrentPr = pullRequest?.approvals?.total || 0;
+  const approvalsCurrentPr = currentPullRequest?.approvals?.total || 0;
   const prsNeedsApproval = approvalsCurrentPr < approvalsRequired;
 
+  async function updateBountyAndPullRequestData({ comments = false, reviews = false, details = false }) {
```

Comment:
this function can be separated into 4:

`updateBountyData(), updatePrDetails(), updatedPrComments(), updatePrReviews()`